third_party/nvfuser/csrc/scheduler/normalization_utils.cpp

```diff
@@ -565,100 +565,73 @@ bool hasSharedInput(
   return has_shared_input;
 }
 
-std::unordered_set<TensorView*> getAllProducerTvsOf(
-    const std::unordered_set<TensorView*>& tv_set,
-    bool skip_reduction_tv) {
-  std::unordered_set<TensorView*> all_producer_tvs;
-  std::queue<TensorView*> unvisited;
-
-  // start search from tv_set, save its direct producers
-  for (auto tv0 : tv_set) {
-    for (auto tv : ir_utils::producerTvsOf(tv0)) {
-      bool skip_tv = skip_reduction_tv && tv->hasReduction();
-      if (!skip_tv && all_producer_tvs.find(tv) == all_producer_tvs.end()) {
-        all_producer_tvs.emplace(tv);
-        unvisited.push(tv);
-      }
+std::unordered_set<TensorView*> getAllTvsFrom(
+    const std::vector<TensorView*>& from_tvs,
+    const std::unordered_set<TensorView*>& cutoff_tv_set,
+    const bool cut_off_reduction_tv) {
+  std::unordered_set<TensorView*> tv_group;
+  std::queue<TensorView*> tensors_to_visit;
+  auto addIfNotVisited = [&](TensorView* tv) {
+    if (tv_group.find(tv) == tv_group.end() &&
+        cutoff_tv_set.find(tv) == cutoff_tv_set.end()) {
+      tv_group.emplace(tv);
+      tensors_to_visit.push(tv);
     }
+  };
+
+  for (auto tv : from_tvs) {
+    tensors_to_visit.push(tv);
   }
-  // search for indirect producers from tv0's direct producers
-  while (!unvisited.empty()) {
-    auto next_tv = unvisited.front();
-    unvisited.pop();
-    for (auto tv : ir_utils::producerTvsOf(next_tv)) {
-      bool skip_tv = skip_reduction_tv && tv->hasReduction();
-      if (!skip_tv && all_producer_tvs.find(tv) == all_producer_tvs.end()) {
-        all_producer_tvs.emplace(tv);
-        unvisited.push(tv);
+  while (!tensors_to_visit.empty()) {
+    auto next_tv = tensors_to_visit.front();
+    tensors_to_visit.pop();
+    // visit consumers
+    for (auto tv : ir_utils::consumerTvsOf(next_tv)) {
+      addIfNotVisited(tv);
+    }
+    // visit siblings
+    for (auto tv : ir_utils::siblingTvsOf(next_tv)) {
+      addIfNotVisited(tv);
+    }
+    // don't visit producer of reduction tv if cut_off_reduction_tv is true
+    if (!next_tv->hasReduction() || !cut_off_reduction_tv) {
```

Comment:
The test depends on `next_tv`, if move into the lambda function, needs to pass next_tv. I modified to the following version:
```
    // visit producer
    if(next_tv->hasReduction() && !scheduler_utils::isFastestDimReduction(next_tv)) {
      // skip producer of outer reduction tv, since they are computed with inner reduction
      continue;
    }else{
      for (auto tv : ir_utils::producerTvsOf(next_tv)) {
        addIfNotVisited(tv);
      }
    }
```